lib/r10k/action/runner.rb

```diff
@@ -44,10 +44,13 @@ def setup_settings
 
         overrides = {}
         overrides[:cachedir] = @opts[:cachedir] if @opts.key?(:cachedir)
-        overrides[:deploy] = {} if @opts.key?(:'puppet-path') || @opts.key?(:'generate-types')
-        overrides[:deploy][:puppet_path] = @opts[:'puppet-path'] if @opts.key?(:'puppet-path')
-        overrides[:deploy][:puppet_conf] = @opts[:'puppet-conf'] unless @opts[:'puppet-conf'].nil?
-        overrides[:deploy][:generate_types] = @opts[:'generate-types'] if @opts.key?(:'generate-types')
+        if @opts.key?(:'puppet-path') || @opts.key?(:'generate-types') || @opts.key?(:'deploy-spec')
```

Comment:
Why isn't `:'puppet-conf'` included in this check? Do we not want to set that unless one of these other options are also present??

Reply:
I just wanted this to be a little bit clearer but the same logic + the bit about `deploy_spec`; I don't know if there is a reason we don't check `:puppet-conf`. 🤷

Reply:
Honestly, seems like it was a bug. I would add it, this will error if you specify _only_ `puppet-conf` right now.

Reply:
Or rather, before it would have errored. Now it will no-op.

Reply:
yeah, that's what it seemed like to me